sdk/azcore/runtime/pager.go

```diff
@@ -0,0 +1,93 @@
+//go:build go1.18
+// +build go1.18
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package runtime
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/internal/shared"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+)
+
+// PageProcessor contains the required data for constructing a Pager.
+type PageProcessor[T any] struct {
+	// Do sends the request to fetch the next page.
+	Do func(*policy.Request) (*http.Response, error)
```

Comment:
We might also be able to do some further consolidation here, get rid of `Do` and merge `Fetcher` and `Responder`.  Let me try it out.

Reply:
This does work.  It makes creating fetchers a bit more complex but is likely worth it in the long run.  Plus, it removes the hard-coded HTTP status code check which bugged me.